src/processor/intents/spawns/_charge-energy.js

```diff
@@ -4,10 +4,28 @@ var _ = require('lodash'),
     C = driver.constants;
 
 
-module.exports = function(spawn, roomObjects, cost, bulk, roomController) {
-
-    var spawns = _.filter(roomObjects, i => i.type == 'spawn' && i.user == spawn.user && !i.off);
-    var extensions = _.filter(roomObjects, i => i.type == 'extension' && i.user == spawn.user && !i.off);
+module.exports = function(spawn, roomObjects, cost, bulk, energyStructures) {
+
+    var spawns = [];
+    var extensions = [];
```

Comment:
Could be a single list of energy buildings now.

Reply:
It would be nice to do so and would simplify the code. But doing so would not retain the exact functionality currently in use with createCreep.

It's @artch decision. I can change it to use a single list of energy buildings. The effect for current createCreep function would be that spawns are no longer preferred over extensions. Buildings would be used in the order of comparatorDistance regardless of type.